scripts/ci/docker-compose-local.yml

```diff
@@ -0,0 +1,42 @@
+#
+# Licensed to the Apache Software Foundation (ASF) under one or more
+# contributor license agreements.  See the NOTICE file distributed with
+# this work for additional information regarding copyright ownership.
+# The ASF licenses this file to You under the Apache License, Version 2.0
+# (the "License"); you may not use this file except in compliance with
+# the License.  You may obtain a copy of the License at
+#
+#    http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+version: "2.2"
+services:
+  airflow-testing:
+    volumes:
+      - ../../airflow:/opt/airflow/airflow
+      - ../../setup.cfg:/opt/airflow/setup.cfg
+      - ../../setup.py:/opt/airflow/setup.py
+      - ../../dags:/opt/airflow/dags
+      - ../../dev:/opt/airflow/dev
+      - ../../docs:/opt/airflow/docs
+      - ../../hooks:/opt/airflow/hooks
+      - ../../scripts:/opt/airflow/scripts
+      - ../../tests:/opt/airflow/tests
+      - ../../.coveragerc:/opt/airflow/.coveragerc
+      - ../../LICENSE:/opt/airflow/LICENSE
+      - ../../MANIFEST.in:/opt/airflow/MANIFEST.in
+      - ../../NOTICE:/opt/airflow/NOTICE
+      - ../../CHANGELOG.txt:/opt/airflow/CHANGELOG
+      - ../../.github:/opt/airflow/.github
+      - ../../.flake8:/opt/airflow/.flake8
+      - ../../pylintrc:/opt/airflow/pylintrc
+      - ../../.rat-excludes:/opt/airflow/.rat-excludes
+      - ../../run-tests:/opt/airflow/run-tests
+      - ../../logs:/root/airflow/logs
```

Comment:
Was there a reason you did it this way this instead of

```
- ../../:/opt/airflow
- ../../logs:/root/airflow/logs
```

(For example, this doesn't mount `.mypy_cache/` which will slow down running mypy the first time in each container. Not the end of the world, but mounting it may be an idea if we keep it this way)

Reply:
Yeah. The main reason is that we hit an interesting problem during testing (wiht Stefan). One of the fields in the "airflow" dir after running `pip install -e .` is apache_airflow.egg-info. And the problem was when we removed werkzeug. Even if it was missing in setup.py, when you entered the docker environment the "host" apache_airflow.egg-info still contained werkzeug and running "airflow" binary failed. It was a kind of mind-blowing problem (no Werkzeug in any of the sources, docker file rebuild from scratch but still Werkzeug was there). 

I've added .mypy_cache indeed :).

Reply:
Oh good to know.